### ISSUE TITLE
Unable to set Additional Settings while setting up a Kinesis trigger to Lambda using aws_lambda_event_source_mapping

### ISSUE THREAD

**Issue by vidyakacquia** [terraform-providers/terraform-provider-aws#11386]
### Community Note

* Please vote on this issue by adding a 👍 [reaction](https://blog.github.com/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/) to the original issue to help the community and maintainers prioritize this request
* Please do not leave "+1" or other comments that do not add relevant new information or questions, they generate extra noise for issue followers and do not help prioritize the request
* If you are interested in working on this issue or have submitted a pull request, please leave a comment

### Description

AWS provides the "Additional Settings" option while creating a Kinesis trigger to Lambda as below:
<img width="859" alt="Screen Shot 2019-12-20 at 20 51 14" src="https://user-images.githubusercontent.com/44468598/71264647-8dc19100-236a-11ea-954c-fc26879ea6ca.png">

There are no documents available in Terraform, which describes the config to include the same. The goal is to include the option provided by AWS. 

### New or Affected Resource(s)

**aws_lambda_event_source_mapping**

### Potential Terraform Configuration

```
resource "aws_cloudwatch_event_target" "event_target" {
  target_id  = var.target_id
  depends_on = [aws_lambda_function.Kinesislambdaprocess]
  rule       = aws_cloudwatch_event_rule.cw_event_rule.name
  arn        = aws_lambda_function.Kinesislambdaprocess.arn
 additional_settings {
    retry_attempts = 3
    maximum_age_of_record = 1000
 }
}
```

### References
Could not find any.

**Comment by bflad** [comment]
It appears that while we do not have a feature request for this at the moment (this issue can serve as that), there is an open pull request that should cover this functionality here: https://github.com/terraform-providers/terraform-provider-aws/pull/11100 👍

**Comment by bflad** [comment]
Support for the additional `aws_lambda_event_source_mapping` arguments has been merged and will release with version 2.51.0 of the Terraform AWS Provider, end of next week. Thanks to @kostas-theo for the implementation. 👍
